framework/include/actions/SetupMeshAction.h

```diff
@@ -35,4 +35,7 @@ class SetupMeshAction : public MooseObjectAction
   const bool _use_split;
   /// The split mesh file (if any); comes from either Mesh/split_file or --split-file
   const std::string _split_file;
+
+  /// Whether to ignore mesh generators and use FileMesh when both are present in input
+  bool _file_mesh_supersedes_mesh_generators;
```

Comment:
Seeing as how there are more non-Mesh Generator object types out there beside `FileMesh`, it might be good to generalize this variable name to something like `_are_mesh_generators_superseded`. I'll let the reviewer weigh in on this.